Add use_auth to load_datasets for private datasets to PT and TF examples

KMFODA opened the pull request:
# What does this PR do?

As per https://github.com/huggingface/transformers/issues/16235 this PR adds the capability to run examples that use trainer on private datasets by adding the use_auth variable to the load_datasets function.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

@sgugger

KMFODA committed:
fix formatting and remove use_auth

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_16521). All of your documentation changes will be reflected on that endpoint.

sgugger reviewed (approved):
Thanks a lot for your PR! Do you have time to add the same changes in the Flax and TensorFlow examples as well?

KMFODA committed:
Add use_auth_token to Flax examples

KMFODA commented:
@sgugger sure thing! I had already added for TF and have just added for Flax. Some examples didn't have `use_auth_token` as a model argument so I added that as well.

sgugger commented:
Thanks a lot for all your work on this!
